mmv1/third_party/terraform/resources/resource_container_cluster.go.erb

```diff
@@ -1713,6 +1713,23 @@ func resourceContainerCluster() *schema.Resource {
 					},
 				},
 			},
+			"gateway_api_config": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				MaxItems:    1,
+				Description: `Configuration for GKE Gateway API controller.`,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"channel": {
+							Type:         schema.TypeString,
+							Default:      "CHANNEL_UNSPECIFIED",
```

Comment:
Generally we prefer to default to an empty string / not sending a value rather than defaulting to CHANNEL_UNSPECIFIED - is this required?

If the API sets a default for this value when it's not specified, you could instead make this field `Optional` and `Computed`.
If the API just does nothing when this value isn't specified, you could make it `Optional` with no default.

```suggestion
```

Reply:
I checked the Golang doc and it will be omitted if empty and the CHANNEL_UNSPECIFIED will be used instead by default. So you are right, we do not need to define it in the provider code! I will remove the default.

https://pkg.go.dev/google.golang.org/api/container/v1#GatewayAPIConfig